Ensure correct MapNumber logic when restoring from backups with draws + add series clinch config parameter + fix OnSeriesResult

nickdnk opened the pull request:
This should fix https://github.com/splewis/get5/issues/817.

I also adjusted `Get5_OnBackupRestore` to have the correct map and round number, indicating the match, map and round being restored **to**.

Tried to reduce use of `Get5_GetMapNumber()` where it is not needed.

Fixed problems with ties not correctly affecting the next map/series end logic.

I put the fix for https://github.com/splewis/get5/issues/822 in here as well.

Added a `g_SeriesCanClinch` option that allows you to disable clinch-logic in a series, similarly to `mp_match_can_clinch`, but for maps.

I've moved a lot of code around, fixing the delay on `OnSeriesResult` and probably fixing a lot of weird issues with GOTV recordings. `mp_match_restart_delay` is now corrected if it is not long enough, but not shortened if longer. This allows admins to set a longer, fixed pause length between maps regardless of the length of the GOTV delay. Essentially this now includes https://github.com/splewis/get5/pull/825, but I had to put that on top of this due to the way map scores are now handled. This also gets rid of `g_ForceWinnerSignal` and `g_ForcedWinner`.

Everything here is tested and it all works as far as I can tell.

nickdnk commented on scripting/get5/backups.sp:
```diff
@@ -292,11 +300,13 @@ public bool RestoreFromBackup(const char[] path) {
   if (kv.JumpToKey("valve_backup")) {
     g_SavedValveBackup = true;
     kv.ExportToFile(tempValveBackup);
+    roundNumberRestoredTo = kv.GetNum("round", 0) - 1; // Round number is not zero-indexed inside valve backup
```
Need to verify that this is correct; that the round number actually matches the one we restored to.

nickdnk commented on scripting/get5/matchconfig.sp:
```diff
@@ -104,19 +104,21 @@ stock bool LoadMatchConfig(const char[] config, bool restoreBackup = false) {
+    if (!restoreBackup) {
+      // When restoring from backup, changelevel is called after loading the match config.
+      // In all other cases, we want to load map 0 in the series.
+      g_MapPoolList.GetString(0, mapName, sizeof(mapName));
```
We might not be able to put 0 here. If someone has `series_score` set in their match config, we shouldn't load map 0, should we? If it starts 1-0, we load map 1, skipping the first one?

nickdnk commented:
Last commit fixes https://github.com/splewis/get5/issues/728

nickdnk commented:
Tested everything in here and it seems to work as expected. Tested `clinch_series` and also tested that bo2 still works as expected.

nickdnk commented on scripting/get5.sp:
```diff
@@ -1150,9 +1151,6 @@ public Action Event_MatchOver(Event event, const char[] name, bool dontBroadcast
     Stats_ResetGrenadeContainers();
 
-    // Write backup before series score increments
-    WriteBackup();
```
I took this out. Per https://github.com/splewis/get5/issues/822 I don't understand why we do it.

splewis replied:
I don't recall any reason for this; I'm okay with removing it. (Worst case, we discover a bug and can actually document it, but I trust your testing, so I don't expect any issue.)

nickdnk commented:
Now also fixes https://github.com/splewis/get5/issues/213

nickdnk commented on scripting/get5.sp:
```diff
@@ -1650,11 +1679,6 @@ public Action Timer_PostKnife(Handle timer) {
   EnsureIndefiniteWarmup();
 }
 
-public Action StopDemo(Handle timer) {
```
This method was not used anywhere.

PhlexPlexico replied:
Couldn't we use this method though as a helper for #381?

nickdnk replied:
We already have a helper. I created a new file called `recording.sp` and put everything GOTV-related into that:

https://github.com/splewis/get5/pull/821/files#diff-bd010f45d65060a909e17e0429128077905e24c818c5d4d9a85cc4cc83702571R1-R24

nickdnk commented:
Fixes https://github.com/splewis/get5/issues/190 now also.

nickdnk commented:
I changed the translation logic to always inject the color around variables if applicable. This means we don't have to make sure different languages use the same colors. I also cleaned up some of the translations a bit, where I could.

Player names are now colored based on the side they are on, so for instance when `PlayerName has requested a tactical pause.` is printed, `PlayerName` will now be either yellow or blue, depending on the side the player is on.

Commands in translations (.ie. `type !unpause to continue`) are now always colored green and have been changed to be injected variables as well.

nickdnk commented:
I changed the translation logic to always inject the color around variables if applicable. This means we don't have to make sure different languages use the same colors. I also cleaned up some of the translations a bit, where I could.

Player names are now colored based on the side they are on, so for instance when `PlayerName has requested a tactical pause.` is printed, `PlayerName` will now be either yellow or blue, depending on the side the player is on.

Commands in translations (.ie. `type !unpause to continue`) are now always colored green and have been changed to be injected variables as well.

splewis commented on scripting/get5/recording.sp:
```diff
@@ -0,0 +1,124 @@
+stock bool StartRecording() {
```
nit: it probably makes more sense for this & StopRecording to be a `public` still (though practically it doesn't really matter). The only value of stock is that the function wouldn't get compiled into the plugin if unused (so it's typically reserved for "library" style files included by many plugins).

splewis commented on scripting/get5/recording.sp:
```diff
@@ -0,0 +1,124 @@
+}
+
+public Action Timer_StopGoTVRecording(Handle timer, DataPack pack) {
+
```
Superfluous blank line.

splewis commented on scripting/get5/recording.sp:
```diff
@@ -0,0 +1,124 @@
+}
+
+public Action Timer_FireStopRecordingEvent(Handle timer, DataPack pack) {
+
```
Superfluous blank line.

splewis reviewed (approved):
Looks good overall, just a few tiny nits that I'd like to see tweaked, feel free to merge it.
